core/nodepool/config/templates/cloud-config-worker

```diff
@@ -1585,5 +1658,33 @@ write_files:
           fi
       done
 
+{{ end }}
+{{ if .EC2Instance.HasNvmeDevices }}
+  - path: /opt/nvme-hack/ebs-nvme-mapping
+    owner: root:root
+    permissions: 0755
+    content: |
+      #! /bin/bash
+      PATH="${PATH}:/usr/sbin"
+
+      for blkdev in $( nvme list | awk '/^\/dev/ { print $1 }' ) ; do
+        mapping=$(nvme id-ctrl --raw-binary "${blkdev}" | cut -c3073-3104 | tr -s ' ' | sed 's/ $//g')
+        if [[ "${mapping}" == /dev/* ]]; then
+          ( test -b "${blkdev}" && test -L "${mapping}" ) || ln -s "${blkdev}" "${mapping}"
+        else
+          mapping="/dev/${mapping}"
+          if [[ "${mapping}" == /dev/* ]]; then
+            ( test -b "${blkdev}" && test -L "${mapping}" ) || ln -s "${blkdev}" "${mapping}"
+          fi
+        fi
+      done
+
+  - path: /opt/nvme-hack/99-aws-ebs-nvme.rules
+    owner: root:root
+    permissions: 0644
+    content: |
+      # Create the legacy device node for the volume mount and raid0 startup
+      ACTION=="add", SUBSYSTEM=="block", KERNEL=="nvme[1-26]n1", ATTRS{model}=="Amazon Elastic Block Store              ", RUN+="/opt/nvme-hack/ebs-nvme-mapping"
```

Comment:
Oh! Thank you so much for the extra!
I believe this feature isn't the primary topic of this PR but I appreciate this anyway.
Would you mind clarifying how this implementation differs from https://github.com/coreos/bugs/issues/2399#issuecomment-379730681?
Due to my lack of expertise, I'm not sure which one is the complete solution.

Reply:
The services seem about the same.

The one I provide tries to map all it can find (via `nvme list`), while the one in [coreos/bugs#2399 (comment)](https://github.com/coreos/bugs/issues/2399#issuecomment-379730681)
looks like it needs to be explicitly invoked for each presumed device to get the mappings.

I've also seen evidence of the `nvme id-ctrl` output both include and exclude a leading "/dev/", so my version
looks for both and maps them as discovered.

I based my effort on [github.com/oogali/ebs-automatic-nvme-mapping](https://github.com/oogali/ebs-automatic-nvme-mapping), which I had been pointed at by our AWS support people.